### PR TITLE
fix: Use Windows-safe filenames (i.e. no colon character)

### PR DESCRIPTION
**Pull request by smockle**
### Problem

Follow-up to https://github.com/github/accessibility/pull/2538

This action produces filenames (e.g. `2022-06-10T18:34:44.775Z.md`) containing colon (`:`) characters, which are disallowed by NTFS (Windows’ default filesystem). When a repo includes filenames with colons, `git clone` and `git checkout` fail on Windows.

### Solution

This PR removes the time portion (i.e. the portion which contains colon characters) of the ISO 8601 _date and time_ format used in generated filenames. Now, generated filenames use the ISO 8601 _date_ format; from [“ISO 8601 Date and time format”](https://www.iso.org/iso-8601-date-and-time-format.html):

> ISO 8601 tackles this uncertainty by setting out an internationally agreed way to represent dates:
> 
> YYYY-MM-DD
> 
> For example, September 27, 2012 is represented as 2012-09-27.

### Validation

I ran `npm ci && npm run all` in a Codespace, and all tests passed.

### CODE REVIEW

File: src/commitToBranch.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ mutation myCreateCommitOnBranch($input: CreateCommitOnBranchInput!) {
 `
 
 const commitToBranch = async ({ owner, repo }: InputFields, username: string, branchNodeId: string, branchSha: string, documentBody: string): Promise<{ ref: { id: string } }> => {
-    const now = (new Date()).toISOString();
+    const today = (new Date()).toISOString().replace(/T.*$/, "");
```
Comment (kendallgassner) [inline src/commitToBranch.ts:16]: I wonder if we should just shorten this to MM/DD/YYYY

Comment (kendallgassner) [reply src/commitToBranch.ts:16]: Something like: `new Date().toLocaleDateString(undefined, 'MM-DD-YYYY')`

Comment (smockle) [reply src/commitToBranch.ts:16]: Thanks for the suggestions! I see a couple potential disadvantages:
1. `toLocaleDateString` will give different values on different [runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners), based on their location.
2. `MM/DD/YYYY` (or `MM-DD-YYYY`) results in an odd sort order, placing January 2022 reports next to January 2023 reports, rather than next to February 2022 reports.

I did consider alternative approaches, such as:
```ts
const date = new Date();
const year = date.getUTCFullYear();
const month = String(date.getUTCMonth()+1).padStart(2, '0');
const date = String(date.getUTCDate()).padStart(2, '0');
const today = `${year}-${month}-${date}`
```

But this PRs’s approach seemed simpler to me. `T` (in the regular expression) is something of a [magic string](https://en.wikipedia.org/wiki/Magic_string), which I’d generally avoid, except that in this case it’s specified in ISO 8601.

Comment (kendallgassner) [reply src/commitToBranch.ts:16]: > results in an odd sort order
@smockle good catch!

**Review by kendallgassner — APPROVED** [review]

**Review by inkblotty — APPROVED** [review]
Thank you for this @smockle 🚀
